wsapi/user.go

```diff
@@ -35,15 +35,9 @@ func (api *API) userTyping(req *model.WebSocketRequest) (map[string]interface{},
 		parentId = ""
 	}
 
-	omitUsers := make(map[string]bool, 1)
-	omitUsers[req.Session.UserId] = true
+	appErr := api.App.PublishUserTyping(req.Session.UserId, channelId, parentId)
```

Comment:
This _might_ be one of the hot areas of the code. Doing two additional DB requests to fetch user and channel could have an high impact. I'm wondering if the readability is really worth the risk of performance issues here.

Reply:
That's a really good point, @hanzei! Yes, I think we should avoid the extra database read here. In fact, since the existing ws code and the new api4 code already do permission checking, it seems entirely redundant to check that the user and channel are valid at all. That is, I think it's fine to allow a plugin to emit values that "don't make sense" in this context: it's not, strictly speaking, wrong: just meaningless.

So maybe we can remove the two database checks altogether?

Reply:
Added a commit https://github.com/mattermost/mattermost-server/pull/14331/commits/20e86f46e4be8a1bf3be982ac134e8bfd5e8112f to remove check. Is this what you had in mind?